### PR TITLE
Migrate Color.toString() test, improves `equalsIgnoringHashCodes`

### PR DESCRIPTION
**Pull request by gaaclarke**
This migrates the last failing test for https://github.com/flutter/engine/pull/54981.  In order to effectively resolve that test I had to make `equalsIgnoringHashCodes` more usable by printing out the line that differs instead of just a huge blob of "expected" vs "actual.

## example
Here's the output after the change.

### test
```
  test('equalsIgnoringHashCodes - wrong line', () {
    expect(
      '1\n2\n3\n4\n5\n6\n7\n8\n9\n10',
      equalsIgnoringHashCodes('1\n2\n3\n4\n5\n6\na\n8\n9\n10'),
    );
  });
```
### output
```
Expected: normalized value matches
          '1\n'
            '2\n'
            '3\n'
            '4\n'
            '5\n'
            '6\n'
            'a\n'
            '8\n'
            '9\n'
            '10'
  Actual: '1\n'
            '2\n'
            '3\n'
            '4\n'
            '5\n'
            '6\n'
            '7\n'
            '8\n'
            '9\n'
            '10'
   Which: Lines 7 differed, expected: 
          'a'
          but got
          '7'
```

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Comment by goderbauer** [comment]
Looks like some checks are unhappy on this one.

**Comment by gaaclarke** [comment]
> Looks like some checks are unhappy on this one.

@goderbauer It was a bad rebase, it missed my last PR that landed somehow.  It should be good now.

**Review by goderbauer — COMMENTED** [review]
LGTM, but can you add a test to test the new matcher output?

**Comment by gaaclarke** [comment]
> LGTM, but can you add a test to test the new matcher output?

None of the output for matchers are tested because there's no way to grab the text.  I think we'd have to make the Matcher exported in order to be able to test the output text.  I figured that was by design.  Let me know if there is another way.

**Comment by gaaclarke** [comment]
> LGTM, but can you add a test to test the new matcher output?

I found a way to do it by catching the exception, done.

File: packages/flutter_test/test/matchers_test.dart [hunk 1]
```diff
@@ -194,6 +194,26 @@ void main() {
...
+    }
+
+    expect(failure, isNotNull);
+    if (failure != null) {
+      final String? message = failure.message;
+      expect(message, contains('Lines 7 differed'));
+      expect(message, contains("'a'"));
+      expect(message, contains("'7'"));
+    }
+  });
```
Comment (jonahwilliams) [inline packages/flutter_test/test/matchers_test.dart:215]: there is also a way to do this with test matchers FYI

```
expect(() => methodThatThrows()), throwsA<TestFailure>()
 .having(..put prop checks here..)

```

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Pull request merged**